sdk/healthinsights/health-insights-radiologyinsights-rest/review/health-insights-radiologyinsights.api.md

```diff
@@ -0,0 +1,1090 @@
+        name: string;
+        type?: CodeableConcept;
+    }[];
+    partOf?: Array<Reference>;
+    period?: Period;
+    phase?: CodeableConcept;
+    primaryPurposeType?: CodeableConcept;
+    principalInvestigator?: Reference;
+    protocol?: Array<Reference>;
+    reasonStopped?: CodeableConcept;
+    resourceType: "ResearchStudy";
+    site?: Array<Reference>;
+    sponsor?: Reference;
+    status: string;
+    title?: string;
+}
+
+// @public
+export interface ResearchStudyOutput extends DomainResourceOutputParent {
+    arm?: {
+        name: string;
+        type?: CodeableConceptOutput;
+        description?: string;
+    }[];
+    category?: Array<CodeableConceptOutput>;
+    condition?: Array<CodeableConceptOutput>;
+    contact?: Array<ContactDetailOutput>;
+    description?: string;
+    enrollment?: Array<ReferenceOutput>;
+    focus?: Array<CodeableConceptOutput>;
+    identifier?: Array<IdentifierOutput>;
+    keyword?: Array<CodeableConceptOutput>;
+    location?: Array<CodeableConceptOutput>;
+    note?: Array<AnnotationOutput>;
+    objective?: {
+        name: string;
+        type?: CodeableConceptOutput;
+    }[];
+    partOf?: Array<ReferenceOutput>;
+    period?: PeriodOutput;
+    phase?: CodeableConceptOutput;
+    primaryPurposeType?: CodeableConceptOutput;
+    principalInvestigator?: ReferenceOutput;
+    protocol?: Array<ReferenceOutput>;
+    reasonStopped?: CodeableConceptOutput;
+    resourceType: "ResearchStudy";
+    site?: Array<ReferenceOutput>;
+    sponsor?: ReferenceOutput;
+    status: string;
+    title?: string;
+}
+
+// @public
+export interface Resource extends Record<string, unknown> {
+    id?: string;
+    implicitRules?: string;
+    language?: string;
+    meta?: Meta;
+    resourceType: string;
+}
+
+// @public
+export interface ResourceOutput extends Record<string, any> {
+    id?: string;
+    implicitRules?: string;
+    language?: string;
+    meta?: MetaOutput;
+    resourceType: string;
+}
+
+// @public (undocumented)
+export interface Routes {
+    (path: "/radiology-insights/jobs/{id}", id: string): GetJob;
+    (path: "/radiology-insights/jobs"): CreateJob;
+}
+
+// @public
+export interface SampledData extends Element {
+    data?: string;
+    dimensions: number;
+    factor?: number;
+    lowerLimit?: number;
+    origin: Quantity;
+    period: number;
+    upperLimit?: number;
+}
+
+// @public
+export interface SampledDataOutput extends ElementOutput {
+    data?: string;
+    dimensions: number;
+    factor?: number;
+    lowerLimit?: number;
+    origin: QuantityOutput;
+    period: number;
+    upperLimit?: number;
+}
+
+// @public
+export interface SexMismatchInferenceOutput extends RadiologyInsightsInferenceOutputParent {
+    kind: "sexMismatch";
+    sexIndication: CodeableConceptOutput;
+}
+
+// @public
+export interface TimePeriod {
+    end?: Date | string;
+    start?: Date | string;
+}
+
+// @public
+export interface TimePeriodOutput {
+    end?: string;
```

Comment:
Here is actually the opposite - should we start with `string | Date`? It'll be a breaking change to expand an output parameter

Reply:
Yeah, the scenario with the output type is the opposite. The service returns data as a `string`, and without implementing a runtime mapping between wire and client types, converting `string` to `Date` directly can be risky. Attempting to automatically parse strings that resemble dates could introduce performance issues and isn't reliable— not every string formatted like a date should be converted to a `Date` object. Therefore, we restrict the output type to `string` in line with JSON types for the RLC outputs, prioritizing safety and performance.

An important principle of RLCs is that they serve as a direct representation of the REST API, complemented by types, without introducing additional abstractions that require execution at runtime.